pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
```

Comment:
Sorry, commenting late.

First of all: All fields in v1beta1 and v1beta2 need description tags. This will also soon be true for v1beta3. The Travis check was broken, but was fixed today. Please ensure future fields have descriptions.

As discussed in the recent PR to update api-conventions.md, it is acceptable for an object to not distinguish Spec and Status if we're confident it will only support one or the other, as in this case.

As discussed in #1627 and #1553, we will want a very similar object for dynamic configuration distribution. But, similar to the distinction between labels and annotations, I agree that it's useful to separate the two use cases.